models/user.py

```diff
@@ -34,11 +39,35 @@ def add_course(self, course):
             self.course_keys.append(course.key)
             self.put()
 
+    def create_project(self, project, course):
+        """Create a project for a student and make student a member"""
+        if project:
+            # Add the student as project member
+            project.add_member(self)
+            # Add the project to the student
+            self.project_keys.append(project.key)
+            if course:
+                # Add the course the project is for
+                project.course = course.key
```

Comment:
You'll want to save the project after updating it, otherwise you'll loose the fact that we've joined it with a course.  I would say that this really isn't the place to create this relationship, though; it makes more sense to just have the course key set when the project object is created instead of doing it inside this method.